File: src/main/java/models/Categorie.java
package models;

public class Categorie {
    int id;
    String nom;
    String description;
    String logo;
    int nbr_partenaire;

    public Categorie() {}
    public Categorie( int id,String nom, String description, String logo, int nbr_partenaire) {
        this.id=id;
        this.nom = nom;
        this.description = description;
        this.logo = logo;
        this.nbr_partenaire = nbr_partenaire;

    }
    public Categorie( String nom, String description,String logo) {

        this.nom = nom;
        this.description = description;
        this.logo = logo;
    }

    public Categorie( String nom, String description, String logo, int nbr_partenaire) {
        this.nom = nom;
        this.description = description;
        this.logo = logo;
        this.nbr_partenaire = nbr_partenaire;

    }


    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getLogo() {
        return logo;
    }

    public void setLogo(String logo) {
        this.logo = logo;
    }

    public int getNbr_partenaire() {
        return nbr_partenaire;
    }

    public void setNbr_partenaire(int nbr_partenaire) {
        this.nbr_partenaire = nbr_partenaire;
    }

    @Override
    public String toString() {
        return "Categorie{" +
                "id=" + id +
                ", nom='" + nom + '\'' +
                ", description='" + description + '\'' +
                ", logo='" + logo + '\'' +
                ", nbr_partenaire=" + nbr_partenaire +
                '}';
    }

}

File: ProjetWebTrekSwap/src/Entity/Categorie.php
<?php

namespace App\Entity;

use App\Repository\CategorieRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: CategorieRepository::class)]
class Categorie
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "Le nom de la catégorie est obligatoire.")]
    #[Assert\Length(
        min: 3,
        max: 100,
        minMessage: "Le nom doit comporter au moins {{ limit }} caractères.",
        maxMessage: "Le nom ne peut pas dépasser {{ limit }} caractères."
    )]
    #[Assert\Regex(
        pattern: "/^(?!\d+$).*$/",
        message: "Le nom ne peut pas être composé uniquement de chiffres."
    )]
    private ?string $nom = null;

    #[ORM\Column(length: 500)]
    #[Assert\NotBlank(message: "La description est obligatoire.")]
    #[Assert\Length(
        min: 10,
        max: 500,
        minMessage: "La description doit comporter au moins {{ limit }} caractères.",
        maxMessage: "La description ne peut pas dépasser {{ limit }} caractères."
    )]
    #[Assert\Regex(
        pattern: "/^(?!\d+$).*$/",
        message: "La description ne peut pas être uniquement composée de chiffres."
    )]
    private ?string $description = null;


    #[ORM\Column(length: 255, nullable: true)]
    private ?string $logo = null;

    #[ORM\Column(nullable: true)]
    #[Assert\NotBlank(message: "Le nombre de partenaires est obligatoire.")]
    #[Assert\PositiveOrZero(message: "Le nombre de partenaires doit être positif ou zéro.")]
    private ?int $nbr_partenaire = null;


    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): static
    {
        $this->nom = $nom;
        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): static
    {
        $this->description = $description;
        return $this;
    }

    public function getLogo(): ?string
    {
        return $this->logo;
    }

    public function setLogo(?string $logo): static
    {
        $this->logo = $logo;
        return $this;
    }

    public function getNbrPartenaire(): ?int
    {
        return $this->nbr_partenaire;
    }

    public function setNbrPartenaire(?int $nbr_partenaire): static
    {
        $this->nbr_partenaire = $nbr_partenaire;
        return $this;
    }
}